### PR TITLE
IBM Spectrum Virtulize Family Storages support for Ansible - host

### PR DESCRIPTION
**Pull request by wangpww**
##### SUMMARY
Add the following files into ansible for IBM Spectrum Virtualize family to support host related operations.
/lib/ansible/module_utils/ibm_svc_utils.py
/lib/ansible/modules/storage/ibm/ibm_svc_host.py

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
Spectrum Virtualize storage from IBM

##### ADDITIONAL INFORMATION
N/A
```paste below

```

### CODE REVIEW

File: lib/ansible/module_utils/ibm_svc_utils.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+        # logging setup
+        self._logger = logging.getLogger(self.__class__.__name__)
+        self.debug = self._logger.debug
+        if log_path:
+            logging.basicConfig(level=logging.DEBUG, filename=log_path)
+
+        # Make sure we can connect via the RestApi
+        self.token = self._svc_authorize()
+        self.debug("_connect token=%s", self.token)
```
Comment (Akasurde) [inline lib/ansible/module_utils/ibm_svc_utils.py]: It is better to not to log sensitive information such as tokens.

**Review by Akasurde — CHANGES_REQUESTED** [review]
Need one more person to take a look into this.

File: lib/ansible/module_utils/ibm_svc_utils.py (hunk 1 above)
Comment (wangpww) [reply lib/ansible/module_utils/ibm_svc_utils.py]: Thanks, fixed it

**Comment by wangpww** [comment]
@tzurE This is peng from IBM CSS china team, could you please help to review this PR? Thank you very much.

File: lib/ansible/modules/storage/ibm/ibm_svc_host.py [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+        # Required
+        self.name = self.module.params['name']
+        self.state = self.module.params['state']
+
+        # Optional
+        self.fcwwpn = self.module.params['fcwwpn']
+        self.iscsiname = self.module.params['iscsiname']
+        self.iogrp = self.module.params['iogrp']
+        self.protocol = self.module.params['protocol']
+        self.type = self.module.params['type']
```
Comment (tzurE) [inline lib/ansible/modules/storage/ibm/ibm_svc_host.py]: If those are optional, wouldn't you rather use some safe way to check their value before passing? Or a default value? not all of them have a default value

Comment (wangpww) [reply lib/ansible/modules/storage/ibm/ibm_svc_host.py]: @tzurE Thank you very much, fixed it, please help to review it.

**Comment by wangpww** [comment]
@Akasurde  and @tzurE  Would you please help to review the PR, I deeply appreciate for your help.

**Comment by wangpww** [comment]
@Akasurde and @tzurE Would you please help to review the PR, I deeply appreciate for your help.

**Comment by wangpww** [comment]
@Akasurde  This PR didn't review more than two weeks, Would you please help to give me any suggestion about it? Thank you very much.

**Comment by tzurE** [comment]
Have you added some system tests? Have you checked it with you Storage Array?

**Comment by wangpww** [comment]
@tzurE Yes, we have Jenkins CI to test it and we also used it for IBM block storage CSI XAVI

**Comment by wangpww** [comment]
@tzurE Any update? Thank you very much for your review and help.

**Comment by wangpww** [comment]
Reviewer, please help to review my PR, my customer are waiting for using it. I deeply appreciate your review. @Akasurde @tzurE

**Comment by wangpww** [comment]
Reviewer, please help to review my PR, my customer are waiting for using it. I deeply appreciate your review. @Akasurde @tzurE

**Comment by wangpww** [comment]
Reviewer, please help to review my PR, my customer are waiting for using it. I deeply appreciate your review. @Akasurde @tzurE

**Comment by wangpww** [comment]
@tzurE  Would you please help to approve this PR, it is a long time for this PR, our customers are waiting for it. Thank you very much!

**Comment by Akasurde** [comment]
Hi @wangpww, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/ibm.spectrum_virtualize<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br>

**Pull request closed without merging**